hyperspy/io_plugins/unbcf_fast.pyx

```diff
@@ -0,0 +1,544 @@
+    # for present bruker technology. If it would change - uncomment bellow and recompile.
+    #return ((<uint64_t>pointer[self.offset-1]<<24) & <uint64_t>0xff00000000000000) |\
+    #   ((<uint64_t>self.buffer2[self.offset-2]<<8) & <uint64_t>0xff000000000000) |\
+    #((<uint64_t>self.buffer2[self.offset-3]<<40) & <uint64_t>0xff0000000000) |\
+    return ((<uint64_t>pointer[4]<<32) & <uint64_t>0xff00000000) |\
+           ((<uint64_t>pointer[3]<<24) & <uint64_t>4278190080) |\
+           ((<uint64_t>pointer[2]<<16) & <uint64_t>16711680) |\
+           ((<uint64_t>pointer[1]<<8) & <uint64_t>65280) |\
+             <uint64_t>pointer[0]
+
+
+# datastream class:
+
+@cython.boundscheck(False)
+cdef class DataStream:
+
+    cdef unsigned char *buffer2
+    cdef int size, size_chnk
+    cdef int offset
+    cdef bytes raw_bytes
+    cdef public object blocks  # public - because it is python object
+
+    def __cinit__(self, blocks, int size_chnk):
+        self.size_chnk = size_chnk
+        self.size = size_chnk
+        self.offset = 0
+
+    def __init__(self, blocks, int size_chnk):
+        self.blocks = blocks
+        self.raw_bytes = next(self.blocks)  # python bytes buffer
+        self.buffer2 = <bytes>self.raw_bytes  # C unsigned char buffer
+
+    cdef void seek(self, int value):
+        """move offset to given value.
+        NOTE: it do not check if value is in bounds of buffer!"""
+        self.offset = value
+
+    cdef void skip(self, int length):
+        """increase offset by given value,
+        check if new offset is in bounds of buffer length
+        else load up next block"""
+        if (self.offset + length) > self.size:
+            self.load_next_block()
+        self.offset = self.offset + length
+
+    cdef uint8_t read_8(self):
+        if (self.offset + 1) > self.size:
+            self.load_next_block()
+        self.offset += 1
+        return <uint8_t>self.buffer2[self.offset-1]
+
+    cdef uint16_t read_16(self):
+        if (self.offset + 2) > self.size:
+            self.load_next_block()
+        self.offset += 2
+        # endianess agnostic way... probably very slow:
+        return read_16(&self.buffer2[self.offset-2])
+
+    cdef uint32_t read_32(self):
+        if (self.offset + 4) > self.size:
+            self.load_next_block()
+        self.offset += 4
+        # endianess agnostic way... probably very slow:
+        return read_32(&self.buffer2[self.offset-4])
+
+    cdef uint64_t read_64(self):
+        if (self.offset + 8) > self.size:
+            self.load_next_block()
+        self.offset += 8
+        return read_64(&self.buffer2[self.offset-8])
+
+    cdef unsigned char *ptr_to(self, int length):
+        """get the pointer to the raw buffer,
+        making sure the array have the required length
+        counting from the offset, increase the internal offset
+        by given length"""
+        if (self.offset + length) > self.size:
```

Comment:
Can `length` be larger than the size of a single block? If so, you should change this to:
```python
while (self.offset + length) > self.size:
```